docs/design/Cephfs-RDR-ConsistencyGroup.md

```diff
@@ -0,0 +1,411 @@
+
+In step 12, volsync will sync the data in the new restored read-only PVC
+(restored by Ramen) to the remote destination
+
+In step 18, Ramen will first create a ReplicationGroupDestination,
+which include the RD spec for each PVC in the consistency group.
+
+* Ramen Update Status.LastSyncStartTime. Go to the Syncing state.
+  In Syncing state:
+    * Create Or Update manual trigger ReplicationDestination for each PVC
+    * Wait all RDs are successfully to receive the data from source by
+      checking the manual string in spec and status
+    * Get latest image from each RD and update the Status.LatestImages
+      in the ReplicationGroupDestination
+    * Update Status.LastSyncStartTime to nil, and Status.LastSyncTime.
+      Go to Cleanup State
+* In Cleanup Sate:
+    * Update Status.LastSyncStartTime to now to go to Syncing state
+
+### Failover
+
+* RS (ReplicationSource): A VolSync resource designed to manage the source.
+  Each RS is responsible for managing one PVC as the data source.
+* RGS (ReplicationGroupSource): A Ramen resource designed to manage all
+  RS instances in a consistency group.
+* RD (ReplicationDestination): A VolSync resource intended for managing the
+  destination. Each RD handles one PVC as the data destination.
+* RGD (ReplicationGroupDestination): A Ramen resource designed to
+  manage all RD instances in a consistency group.
+
+![RDR Failover](../diagrams/cephfs-cg-rdr-failover.drawio.svg)
+
+This diagram illustrates the design for CephFS Regional-DR with
+Consistency Group failover.
+The modifications have been appended to the bottom of the diagram
+rather than altering the original version.
+
+In the current implementation, during step 6, the application's
+PVCs are restored to the
+latest image from their corresponding Replication Destination.
+
+In this design, we record the image in consistency group in the ReplicationGroupDestination.
+So we need to get the latest image from ReplicationGroupDestination
+instead of each ReplicationDestination.
+
+In step 9, additionally, the ReplicationGroupDestination need to be cleaned.
+
+In step 10, currently, Ramen creates a ReplicationSource for each PVC
+to enable Volsync to create independent volume snapshots for each PVC.
+Subsequently, Volsync restores each independent snapshot to a new
+read-only PVC to synchronize the data from it to the remote destination.
+
+In this design, since volsync doesn't currently support consistency groups,
+Ramen takes on the responsibility of creating volume group snapshots for
+the PVCs within the consistency group and restoring these snapshots to PVCs.
+Subsequently, Volsync synchronizes the data from the PVCs restored by
+Ramen to the remote destination.
+
+So, in step 10, there are some differences compared to the
+current implementation if a consistency group is specified:
+
+* Ramen get the existing VolumeGroupSnapshotClass
+* Ramen creates a ReplicationGroupSource with the schedule
```

Comment:
Ramen will have a scheduler (replacing VolSync scheduler) to manage the scheduling of ReplicationGroupSnapshot requests at schedule intervals. Please ensure that the doc reflects this change.